docs/user-guide/commands.md

```diff
@@ -0,0 +1,105 @@
+---
+layout: main
+title: Commands
+category: User Guide
+menu: menu
+toc:
+    - title: Commands
+      url: "#commands"
+    - title: Using a command
+      url: "#using-a-command"
+    - title: Creating a command
+      url: "#creating-a-command"
+    - title: Finding commands
+      url: "#finding-commands"
+    - title: More links
+      url: "#more-links"
+---
+# Commands
+
+Screwdriver commands are executables which can either be a group of [commands](https://en.wikipedia.org/wiki/Command_(computing)) in a script or a binary that people can use to replace a step definition in a [screwdriver.yaml](./configuration).
+
+## Using a command
+
+To use a command, define a `screwdriver.yaml` that uses the `sd-cmd` cli with the format:
+
+```
+$ sd-cmd exec <namespace>/<name>@<version> <arguments>
+```
+
+__Input:__
+
+- `namespace/name` - the fully-qualified command name
+- `version` - a semver-compatible format or tag
+- `arguments` - passed directly to the underlying format
+
+__Output:__
+
+All debug logs about the command lookup and execution are stored in `$SD_ARTIFACTS_DIR/.sd/commands/namespace/name/version/timestamp.log`
+
+Example:
+```yaml
+jobs:
+    main:
+        requires: [~pr, ~commit]
+        steps:
+            - exec: sd-cmd exec foo/bar@1 -baz sample
+```
+
+Screwdriver will download that binary or script from the Store, make it executable, and run it with the `-baz sample` arguments directly:
+```
+$ /opt/sd/commands/foo/bar/1.0.1/foobar.sh -baz sample
+```
+
+## Creating a command
+
+Publishing and running commands must be done from a Screwdriver pipeline.
+
+### Writing a command yaml
+
+To create a command, create a repo with a `sd-command.yaml` file. The file should contain a namespace, name, version, description, maintainer email, format, and a binary config with a path to file.
+
+Example `sd-command.yaml`:
+
+```yaml
+namespace: foo # Namespace for the command
+name: bar # Command name
+version: 1.0 # Major and Minor version number (patch is automatic)
+description: |
+  Lorem ipsum dolor sit amet.
+maintainer: foo@bar.com # Maintainer of the command
+format: binary # Format the command is in (binary)
+binary:
+    file: ./foobar.sh # Path to script or binary file
+```
+
+### Writing a screwdriver.yaml for your command repo
+
+To validate your command, run the `command-validate` script from the [screwdriver-command-validator](https://github.com/screwdriver-cd/command-validator) npm module. This means the build image must have NodeJS and NPM properly installed to use it.
+
+To publish your command, run the `sd-cmd publish` command in a separate job. `-f` stands for file.
+
+Example `screwdriver.yaml`:
+```yaml
+shared:
+    image: node:6
+jobs:
+    main:
+        requires: [~pr, ~commit]
+        steps:
+            - install: npm install screwdriver-command-validator
+            - validate: ./node_modules/.bin/command-validate -f sd-command.yaml
```

Comment:
we should enhance sd-cmd to have a validate option, rather than asking users to setup a job and nodejs for this.
eg: `sd-cmd validate -f sd-command.yaml`

Reply:
You can add comments on the issue: https://github.com/screwdriver-cd/screwdriver/issues/677

Reply:
That shouldn't block this PR. I think the reason why they use node is because they want to leverage on the `data-schema` to do the validation. We can have a `command/validate` endpoint and have `sd-cmd validate` hit that endpoint. Will add this to the issue